lib/services/platform-service.ts

```diff
@@ -458,16 +458,25 @@ export class PlatformService implements IPlatformService {
 		if (this.$options.availableDevices) {
 			return this.$emulatorPlatformService.listAvailableEmulators(platform);
 		}
+		this.$options.emulator = true;
 		if (this.$options.device) {
 			let info = this.$emulatorPlatformService.getEmulatorInfo(platform, this.$options.device).wait();
 			if (info) {
 				if (!info.isRunning) {
 					this.$emulatorPlatformService.startEmulator(info).wait();
 				}
 				this.$options.device = null;
+			} else {
+				this.$devicesService.initialize({ platform: platform, deviceId: this.$options.device }).wait();
+				let found: Mobile.IDeviceInfo[] = [];
+				if (this.$devicesService.hasDevices) {
```

Comment:
This code is necessary because deployPlatform method calls $devicesService.execute method in order to deploy on all specified devices:
https://github.com/NativeScript/nativescript-cli/blob/master/lib/services/platform-service.ts#L435

The execute method however starts the default emulator when there are no connected devices:
https://github.com/telerik/mobile-cli-lib/blob/9b2d88f4b23247e601fccb35be45c411ea3ddf87/mobile/mobile-core/devices-service.ts#L318 

All this is not expected when executing `tns emulate <platform> --device <device_id>`. The expected result is to fail with the following message:
"Cannot find device with name: <device_id>" as specified by our QAs. Note that this message is different from ERROR_NO_DEVICES message.
This code comes from PR #2412